src/index.ts

```diff
@@ -90,6 +90,20 @@ class NixInstallerAction extends DetSysAction {
   runnerOs: string | undefined;
 
   constructor() {
+    if (platform.getArchOs() === "X64-macOS") {
+      // Holy guacamole this is ugly
+      actionsCore.error(
+        "Determinate Nix Installer no longer supports macOS on Intel. Please migrate to Apple Silicon, and use Nix's built-in Rosetta support to build for Intel. See: https://github.com/DeterminateSystems/nix-src/issues/224",
+      );
+      const sourceTag = inputs.getStringOrUndefined("source-tag");
```

Comment:
What if they're setting source-rev, or source-pr, or any of our other source-..... args? We probably shouldn't override those too.

Reply:
as far as I can tell, absolutely nobody does that for x86_64-darwin:

<img width="1200" height="388" alt="CleanShot 2025-11-09 at 22 53 24@2x" src="https://github.com/user-attachments/assets/d19b9863-99c2-4332-9a58-87b719c4a260" />